src/viam/trajex/totg/trajectory.cpp

```diff
@@ -439,40 +440,123 @@ enum class integration_event : std::uint8_t {
 }
 
 // Searches for a velocity switching point where escape from velocity curve becomes possible.
-// When trapped on the velocity limit curve, we search forward to find where the trajectory
-// can drop below the curve (equation 40). Uses coarse search followed by bisection refinement.
-// See Kunz & Stilman Section VII-B, equations 40-42.
+// Implements both continuous (equation 40) and discontinuous (equations 41-42) cases from Section VII-B.
 // Returns the switching point on velocity curve. If no escape point found, returns end of path.
 // Takes path::cursor by value (cheap copy) to seed forward search from current position.
 [[gnu::pure]] trajectory::phase_point find_velocity_switching_point(path::cursor cursor, const trajectory::options& opt) {
     const arc_length path_length = cursor.path().length();
-    const double step_size = opt.delta.count();
 
-    // Phase 1: Coarse forward search for escape condition (equation 40).
+    std::optional<trajectory::phase_point> discontinuous_switching_point;
+    std::optional<trajectory::phase_point> continuous_switching_point;
+
+    // Phase 1: Check for discontinuous switching points at segment boundaries (equations 41-42).
+    // Section VII-B case 2: f''_i(s) being discontinuous is a necessary condition for
+    // s_ddot_min(s, s_dot_max_vel(s)) being discontinuous.
+    // We need to check ALL boundaries to find the earliest one, not return immediately.
+    auto boundary_cursor = cursor;
+    while (boundary_cursor.position() < path_length) {
+        auto current_segment = *boundary_cursor;
+        const arc_length boundary = current_segment.end();
+
+        // Make sure we are not at the end of the path
+        if (boundary >= path_length) {
+            break;
+        }
+
+        // Only check boundaries ahead of our starting position
+        if (boundary <= cursor.position()) {
+            boundary_cursor.seek(boundary);
+            continue;
+        }
+
+        // Sample geometry from segment ending at boundary
+        const auto q_prime_before = current_segment.tangent(boundary);
+        const auto q_double_prime_before = current_segment.curvature(boundary);
+
+        // Advance cursor to boundary (moves to next segment)
+        boundary_cursor.seek(boundary);
+        auto segment_after = *boundary_cursor;
+
+        // Sample geometry from segment starting at boundary
+        const auto q_prime_after = segment_after.tangent(boundary);
+        const auto q_double_prime_after = segment_after.curvature(boundary);
+
+        // Compute velocity limits on both sides
+        const auto [s_dot_max_acc_before, s_dot_max_vel_before] =
```

Comment:
I think `s_dot_max_acc_before` and `s_dot_max_acc_after` are unused, maybe replace with placeholders.